charts/meteor-pipelines/templates/validate-jupyterhub-image.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: validate-jupyterhub-image
+spec:
+  params:
+   - name: url
+  results:
+   - name: python-runtime
+     description: Python runtime
+   - name: pip-packages
+     description: Packages installed in runtime
+   - name: message
+     description: Messages passed down from task
+  steps:
+    - name: jupyter-book-build
+      image: $(params.url)
+      securityContext:
+        privileged: true
+      script: |
+         #!/usr/bin/env bash
+
+         # Default values for easier handling when error occurs
+         echo -n "" > $(results.message.path)
+         echo -n "[]" > $(results.python-runtime.path)
+         echo -n "[]" > $(results.pip-packages.path)
+
+         # Check Python version and save it to variable
+         echo  "Checking Python version"
+         version=$(python --version)
+         # Check if Python was found, if not fail task
+         if [[ $? != 0 ]]; then
+              echo "Python is not installed" >&2
+              echo "Python unavailable" > $(results.message.path)
+              exit 1
```

Comment:
Let's try not to exit with any other exit code than 0. While I get the logic here, in Tekton any non zero exit state is treated as a failed task. That means we'd get the same "tekton result" if it's an invalid image due to missing packages as well as if something unexpected went wrong with our script (or other things like ImagePullError)... I think it's safer to report errors via the `message` result - if message is populated it means we found a problem with the image. If the image is valid we report empty message.

Let's reserve the "error state" to things like ImagePullError, wdyt?

Reply:
Sure, makes sense :+1: .

Reply:
Resolved in latest commit.